rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -31,16 +34,14 @@ function exit_handler()
   exec 13>&-
   hdfs dfs -put /tmp/pai_yarncontainer_$CONTAINER_ID.log \
     {{{ hdfsUri }}}/Container/{{{ jobData.userName }}}/{{{ jobData.jobName }}}/log/
+  kill -9 $(docker inspect --format="{{.State.Pid}}" $docker_name)
```

Comment:
>$docker_name [](start = 53, length = 12)

I hope it does not return 0 if the docker_name doesn't exist.

Reply:
It won't.
- If there's no error in trap handler, the exit code is the same as the last command before trap handler.
- If there's an error in trap handler, the exit code should be the same as that.

Reply:
I am not talking about exit code. I am wondering What is the result of "docker inspect --format="{{.State.Pid}}" $docker_name" when $docker_name does *not* exist (i.e., already exited)

---
In reply to: [195936120](https://github.com/Microsoft/pai/pull/735#discussion_r195936120) [](ancestors = 195936120)